File: Matiere.ts
import { Seance } from "./Seance";

export class Matiere {
    nom: string;
    nb_paquets: number;
    seance: Seance;


    constructor(nom: string, nb_paquets: number, seance: Seance){
        this.nom = nom;
        this.nb_paquets = nb_paquets;
        this.seance = seance;
    }

    toString(): string {
        return `Matiere [nom=${this.nom}, nb_paquets=${this.nb_paquets}, seance=${this.seance}]`;
    }

    equals(other: Matiere): boolean {
        if (!(other instanceof Matiere)) {
            return false;
        }
        return this.nom === other.nom;
    }
  }

File: backendProjet1/src/main/java/com/example/projetbackend/entities/Matiere.java
package com.example.projetbackend.entities;


import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;

@Entity
@EqualsAndHashCode(of = {"nom"})
@Getter
@Setter

public class Matiere {
	
	@Id
	private String nom;
	private int nbPaquets;
	
	@ManyToOne
    @JoinColumn(name = "seance")
	@JsonIgnore
    private Seance seance;
	
	    
	
}
